### PR TITLE
add bower support

### PR DESCRIPTION
**Pull request by jabranr**
Already added to following endpoints for [bower.io](http://bower.io):

* wpgulp
* wp-gulp
* WPGulp

For info:

``` shell
$ bower info wpgulp
$ bower info wp-gulp
$ bower info WPGulp
```

To install:

``` shell
$ bower install wpgulp
$ bower install wp-gulp
$ bower install WPGulp
```

### CODE REVIEW

**Comment by ahmadawais** [comment]
Thanks for the PR @jabranr , a few questions though

1. How will `Bower` manage the dependencies of `package.json` considering you need both package.json and `gulpfile.js` to run WPGulp? Which is also the reason why I never registered the endpoint?
2. If you have a solution for concern no. 1 then I would like to register those endpoints myself, it looks like you have registered them, could you maybe transfer them or let me know what other way it is possible?

**Comment by jabranr** [comment]
Yes, I figure that it would be too complicated to make every thing work together. I only wanted to give an extra install option but you are right, it is not feasible for this project. I will unregister the endpoints at bower.

**Comment by jabranr** [comment]
@ahmadawais Seems like I cannot unregister as I am not a collaborator to this repo. Can you temporary add me as collaborator so I can unregister it. Cheers

**Comment by ahmadawais** [comment]
@jabranr Sure! Check now [here](https://github.com/ahmadawais/WPGulp/invitations)

**Comment by jabranr** [comment]
@ahmadawais all done! :+1:

**Comment by ahmadawais** [comment]
@jabranr Thanks. Now can you answer 
How will Bower manage the dependencies of `package.json` considering you need both package.json and gulpfile.js to run WPGulp? Which is also the reason why I never registered the endpoint?

**Comment by jabranr** [comment]
Bower won't manage npm's dependencies. It itself is a dependency manager same as npm but for front-end dependencies. Bower more works like dropping in the required dependencies. So consider if WPGulp is a dependency for any project then it can easily be installed in project using Bower (by default at `bower_components/`). This way you would not need to go to repository and download the files manually. 

Although target directory can be changed easily using `.bowerrc` file but then it is not automagic sort of setup. Literally almost same hassle as if we are downloading the files manually and putting them in project. In fact, I did not try it so initially from your response figured that it will make it complicated but I will give it a try and see what sort of setup works well and then perhaps we can re-add this support. ;)

At the moment I think that this might work well:

Considering that WPGulp needs to be at base of the project then we need following `.bowerrc` file to setup the target install directory.

``` yaml
{
  "directory": "."
}
```

Install WPGulp with Bower:

``` shell
$ bower install WPGulp
```

This should install WPGulp in root/base directory of the project. But I guess problem is that it will install as a folder named WPGulp (need to confirm this) where instead it should just drop WPGulp files.

TL;DR, I will test and confirm! :)

**Comment by ahmadawais** [comment]
@jabranr Can you explain what did you confirm `TL;DR, I will test and confirm! :)`?

Also, since package.json won't be added, how will one handle its dependencies?

**Comment by ahmadawais** [comment]
@jabranr Do you know how I can give you the credit for this PR via terminal? I just registered the WPGulp package at Bower repo in #14

**Comment by jabranr** [comment]
@ahmadawais don't worry about that. You already referenced me there so that's OK. Since we have the bower setup now so we can try it and test it as well.

**Comment by ahmadawais** [comment]
@jabranr I am trying to learn how to do that, I mean credit someone on GitHub. 
Quick search on Google didn't turn up anything.

**Comment by ahmadawais** [comment]
@jabranr do check the current setup and tell me how it can be improved, since I don't use Bower.

**Comment by jabranr** [comment]
Sure, I will do that. You need to merge PRs from contributors so it can link to them. Other ways are only to mention them in tickets etc.

**Comment by ahmadawais** [comment]
@jabranr  I am actually looking into how I can assign a contribution to some GitHUb user by my own terminal via Mac OS X. Do you have any insights on that?

**Comment by jabranr** [comment]
@ahmadawais I don't think you can assign contributions across on Github. I never heard this. There may be some possibility of correcting emails that are wrongly attached to commits using git config - but that still does not actually reassign it.

So I tried using it with few options through Bower but it does not really workout well without some good manual work. I tried `.bowerrc` with following options:

``` json
{
  "scripts": {
    "postinstall": "npm install ./bower_components/WPGulp"
  }
}
```

``` json
{
  "cwd": "./bower_components/WPGulp",
  "scripts": {
    "postinstall": "npm install"
  }
}
```
But the commands are triggered with unexpected outcome. Only way I can see is to add a bash script to the project that runs the `npm install` for WPGulp, i.e.

``` shell
# filename install.sh

$ cd bower_components/WPGulp
$ npm install
```

Then following `.bowerrc`:

``` json
{
  "scripts": {
    "postinstall": "./bower_components/WPGulp/install.sh"
  }
}
```

Outputting a correct structure:

``` shell
-- bower_components
 \-- WPGulp
   \-- node_modules
```

But even then this requires updating the paths manually in WPGulp.js. So basically not really helping anything in supposedly "automated workflow".

**Comment by ahmadawais** [comment]
@jabranr Yup, I think there is really no use of this in Bower.

**Comment by ahmadawais** [comment]
What about if it is an NPM pacakge?

**Comment by jabranr** [comment]
It might work there since it can easily get the dependencies but beware that it will still reside in `node_modules/` and will need to either symlinked or moved to project root to get it working.

**Comment by ahmadawais** [comment]
@jabranr I guess I will just contribute to WPCLI and write a curl/wget fetch script to download `gulpfile.js` and `package.json` into the theme/plugin root folder.

**Comment by jabranr** [comment]
I think that is probably the best solution out of all these.

**Comment by jb510** [comment]
Since Bower is EOL and barely maintained, they recommend not using it for new projects.  

We could probably remove Bower and just focus on NPM.

**Comment by ahmadawais** [comment]
@jb510 You're probably right! It's a closed PR :)
